airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/operation/executor/DefaultWriteOperationExecutor.kt

```diff
@@ -0,0 +1,100 @@
+import io.airbyte.protocol.models.v0.AirbyteMessage
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.micronaut.context.annotation.Requires
+import jakarta.inject.Named
+import jakarta.inject.Singleton
+import java.io.BufferedInputStream
+import java.io.ByteArrayOutputStream
+import java.nio.charset.StandardCharsets
+import java.util.concurrent.TimeUnit
+
+private val logger = KotlinLogging.logger {}
+
+@Singleton
+@Named("writeOperationExecutor")
+@Requires(
+    property = ConnectorConfigurationPropertySource.CONNECTOR_OPERATION,
+    value = "write",
+)
+@Requires(env = ["destination"])
+class DefaultWriteOperationExecutor(
+    private val messageConsumer: SerializedAirbyteMessageConsumer,
+    private val shutdownUtils: ShutdownUtils,
+) : OperationExecutor {
+    override fun execute(): Result<AirbyteMessage?> {
+        logger.info { "Using default write operation executor." }
+        try {
+            getInputStream().use { bis ->
+                ByteArrayOutputStream().use { baos ->
+                    consumeWriteStream(messageConsumer, bis, baos)
+                }
+            }
+            return Result.success(null)
+        } catch (e: Exception) {
+            return Result.failure(
+                OperationExecutionException("Failed to write output from connector.", e)
+            )
+        } finally {
+            try {
+                messageConsumer.close()
+            } catch (e: Exception) {
+                logger.warn(e) { "Failed to close consumer." }
+            }
+            shutdownUtils.stopOrphanedThreads(
+                ShutdownUtils.EXIT_HOOK,
+                ShutdownUtils.INTERRUPT_THREAD_DELAY_MINUTES,
+                TimeUnit.MINUTES,
+                ShutdownUtils.EXIT_THREAD_DELAY_MINUTES,
+                TimeUnit.MINUTES,
+            )
+        }
+    }
+
+    fun getInputStream(): BufferedInputStream {
+        return BufferedInputStream(System.`in`)
+    }
+
+    @Throws(Exception::class)
+    fun consumeWriteStream(
+        consumer: SerializedAirbyteMessageConsumer,
+        bis: BufferedInputStream,
+        baos: ByteArrayOutputStream,
+    ) {
+        consumer.start()
+
+        val buffer = ByteArray(8192) // 8K buffer
+        var bytesRead: Int
+        var lastWasNewLine = false
+
+        while ((bis.read(buffer).also { bytesRead = it }) != -1) {
+            for (i in 0 until bytesRead) {
+                val b = buffer[i]
+                if (b == '\n'.code.toByte() || b == '\r'.code.toByte()) {
+                    if (!lastWasNewLine && baos.size() > 0) {
+                        consumer.accept(baos.toString(StandardCharsets.UTF_8), baos.size())
+                        baos.reset()
+                    }
+                    lastWasNewLine = true
+                } else {
+                    baos.write(b.toInt())
+                    lastWasNewLine = false
+                }
+            }
+        }
```

Comment:
Use `java.util.Scanner` along with `nextLine`? In any case don't roll your own.

Reply:
Probably there is a subtle difference. Bis.read blocks when we get records in bursts, also we use a bounded byteArray buffer, scanner.nextLine will continue to buffer until it finds the linefeed char.  also the downstream call relies in bufferedOutStream.size to allocate/block consumption.

Reply:
OK, still, let's not merge this. Either it's just bad code, or it's deliberately and carefully designed in this way but it's lacking commentary to justify its unique special snowflake status by not using a Scanner or an InputStreamReader.

Personally I find the whole memory considerations completely ridiculous in light of the use of `String` and besides it's not like we're anywhere close to supporting gigabyte-sized records anyway.

Reply:
@postamar This is the existing code from the Java implementation of the `IntegrationRunner` ported to Kotlin.  I did not change or introduce this.  This should not hold up the PR.  If you want to take a stab at fixing it, I suggest we do that as a follow on PR as this is pre-existing code.

Reply:
https://github.com/airbytehq/airbyte/pull/36375
Was this really worth arguing over and deferring? I'm puzzled.

Reply:
@postamar It is definitely lacking commentary (predates me) but I found an interesting test which was specifically testing the output of this method against `java.util.Scanner.in..` as you did in your PR as `oldoutput`.  
https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/java/airbyte-cdk/core/src/test/java/io/airbyte/cdk/integrations/base/IntegrationRunnerBackwardsCompatbilityTest.java#L36 I would prefer this as a follow up too with specific

Reply:
> #36375 Was this really worth arguing over and deferring? I'm puzzled.

@postamar In normal circumstances no.  However, this PR and its predecessor have been open for over two weeks.  If we try to fix every ill that exists in the current code that is either a) being converted to Kotlin or b) being moved into the new framework, this will never merge.  I am totally happy to address these as part of a punch list once this gets merged.  Thank you for taking some time to fix this particular issue.

Reply:
can we at least share implementation rather than copy-pasting code? Now that we've converted everything to kotlin, I don't think there's any reason not to do that

Reply:
@stephane-airbyte  Joe has one in [flight](https://github.com/airbytehq/airbyte/pull/36673) changing this. so shouldn't be a blocker for this PR maybe ?

Reply:
@stephane-airbyte @gisripa We should probably wait for @jbfbell 's PR and then do a refactor to extract this logic to a common location that we can use in both places.

Reply:
Sounds good to me. Oh I see, we want to share this logic between old IntegrationRunner and the new µnaut runner. 👍

Reply:
chiming in here that my PR is merged